tn_api_v2_core.yml

```diff
@@ -0,0 +1,948 @@
+        Before you start ensure that:<br  />
+            - IP addresses have been provided to be whitelisted for our servers.<br  />
+            - API username and password have been provided by Trip Ninja.<br  />
+        
+        <h2>Workflow</h2>
+        The following diagram shows the typical data flow of API calls to Trip Ninja for a typical OTA integration.
+        
+        ![Workflow diagram](https://s3.amazonaws.com/tn-api-docs/ota_flowchart.png)
+        
+        <h2>Authentication</h2>
+        Trip Ninja uses Basic Authentication standards. Simply encode your username and password string using base64 and pass it in the
+        authorization headers. See the python example below on how this is done. In the example, the payload and API endpoint url are not
+        shown.
+        
+        Python Example:
+        
+        <pre>
+        import base64
+        auth = base64.b64encode("USERNAME:PASSWORD")
+        headers = {
+            'authorization': "Basic "+ auth,
+        }
+        response = requests.post(url, headers=headers, data=json.dumps(payload))
+        </pre>
+        
+        # Error Codes
+        <table>
+          <tr><td>IE01</td><td>Need minimum 2 destinations</td></tr>
+          <tr><td>IE02</td><td>Duplicate city in list</td></tr>
+          <tr><td>IE09</td><td>Can't run any queries for flight dates in the past</td></tr>
+          <tr><td>IE10</td><td>Incorrect city_type, expected one of ['C', 'A']</td></tr>
+          <tr><td>IE19</td><td>Incorrect region, expected one of 'emea', 'apac', 'americas'</td></tr>
+          <tr><td>IE21</td><td>Wrong currency</td></tr>
+          <tr><td>IE22</td><td>Not a valid cabin option. expected one of 'E' , 'PE', 'SE', 'BC','FC', 'PFC'</td></tr>
+          <tr><td>IE23</td><td>Not a valid alliance. Expected one of *A, *S, *O</td></tr>
+          <tr><td>IE24</td><td>Travellers Type does not meet expected criteria. Should be a list having between 0 and 10 values</td></tr>
+          <tr><td>IE27</td><td>Num_results should be an integer between 50 and 1000</td></tr>
+          <tr><td>IE29</td><td>Invalid Traveller type</td></tr>
+          <tr><td>IE30</td><td>Incorrect value for single_pnr, must be boolean</td></tr>
+        <table>
+
+    x-logo:
+        url: 'https://s3.amazonaws.com/tn-api-docs/trip_ninja_logo.png'
+        altText: Trip Ninja logo
+        href: 'https://www.tripninja.io/'
+
+servers:
+    -   url: https://preprodapi.tripninja.io
+        description: Pre-Production server for development and testing
+
+paths:
+    /get-searches/:
+        post:
+            summary: "Get Searches"
+            description: |
+                This endpoint carries out FareStructure and if applicable Virtual Interlining logic to determine the datasource
+                searches required. The response contains a list of searches to be done by the OTA, which are then passed in the next
+                endpoint (generate-solutions) to get the results.
```

Comment:
think we can add some more OOOOMF

This endpoint carries out FareStructure and if applicable Virtual Interlining logic to determine the optimal searches predicted to provide the best flight search results. 

The response contains a list of searches to be done by the OTA. Once finished the results are then passed in the next endpoint (generate-solutions)

wdyt?

Reply:
will change the first part, feel like second part sounds good as is!